packages/angular/ssr/src/common-engine.ts

```diff
@@ -16,12 +16,28 @@ import {
 import * as fs from 'node:fs';
 import { dirname, resolve } from 'node:path';
 import { URL } from 'node:url';
-import { InlineCriticalCssProcessor } from './inline-css-processor';
+import { InlineCriticalCssProcessor, InlineCriticalCssResult } from './inline-css-processor';
+import {
+  noopRunMethodAndMeasurePerf,
+  printPerformanceLogs,
+  runMethodAndMeasurePerf,
+} from './peformance-profiler';
 
 const SSG_MARKER_REGEXP = /ng-server-context=["']\w*\|?ssg\|?\w*["']/;
 
+export interface CommonEngineOptions {
+  /** A method that when invoked returns a promise that returns an `ApplicationRef instance once resolved or an NgModule. */
+  bootstrap?: Type<{}> | (() => Promise<ApplicationRef>);
+  /** A set of platform level providers for all requests. */
+  providers?: StaticProvider[];
+  /** Enable request performance profiling data collection and printed in the server console. */
```

Comment:
Should we also mention that this is not recommended for production?
```suggestion
  /** Enable request performance profiling data collection and printing the results in the server console. */
```